### PR TITLE
Filter dangling images using image filters.

### PR DESCRIPTION
**Pull request by manugupt1**
Signed-off-by: Manu Gupta <manugupt1@gmail.com>

### CODE REVIEW

File: cmd/nerdctl/images_test.go [hunk 1]
```diff
@@ -116,3 +116,22 @@ LABEL version=0.1`, testutil.CommonImage)
 	base.Cmd("images", "--filter", "reference=busy*:*libc*").AssertOutContains("glibc")
 	base.Cmd("images", "--filter", "reference=busy*:*libc*").AssertOutContains("uclibc")
 }
+
+func TestDanglingImagesFilter(t *testing.T) {
```
Comment (AkihiroSuda) [inline cmd/nerdctl/images_test.go:120]:
```suggestion
func TestImagesFilterDangling(t *testing.T) {
```

Similar tests should have a common prefix

File: pkg/imgutil/imgutil.go [hunk 2]
```diff
@@ -383,11 +384,20 @@ func ParseRepoTag(imgName string) (string, string) {
...
+	DanglingTrue
+	DanglingFalse
+)
+
 type Filters struct {
 	Before    []string
 	Since     []string
 	Labels    map[string]string
 	Reference []string
+	Dangling  Dangling
```
Comment (AkihiroSuda) [inline pkg/imgutil/imgutil.go:400]:
```suggestion
	Dangling  *bool
```

File: pkg/imgutil/imgutil.go [hunk 3]
```diff
@@ -398,7 +408,15 @@ func ParseFilters(filters []string) (*Filters, error) {
 		case 1:
 			return nil, fmt.Errorf("invalid filter %q", filter)
 		case 2:
-			if tempFilterToken[0] == FilterBeforeType {
+			if tempFilterToken[0] == FilterDanglingType {
+				if strings.EqualFold(tempFilterToken[1], "true") {
```
Comment (AkihiroSuda) [inline pkg/imgutil/imgutil.go:412]: Not `EqualFold`

```console
$ docker images -f "dangling=TRUE"
Error response from daemon: invalid filter 'dangling=[TRUE]'
```

Comment (manugupt1) [reply pkg/imgutil/imgutil.go:412]: I was intentional about it as it still does not break backward compatibility. LMK if you still want me to change it.

File: pkg/imgutil/imgutil.go [hunk 4]
```diff
@@ -369,7 +370,7 @@ func ParseRepoTag(imgName string) (string, string) {
 
 	ref, err := refdocker.ParseDockerRef(imgName)
 	if err != nil {
-		logrus.WithError(err).Warnf("unparsable image name %q", imgName)
+		logrus.WithError(err).Debugf("unparsable image name %q", imgName)
```
Comment (AkihiroSuda) [inline pkg/imgutil/imgutil.go:373]: ?

File: pkg/imgutil/imgutil.go (hunk 3 above)
Comment (AkihiroSuda) [reply pkg/imgutil/imgutil.go:412]: Docker seems case-sensitive, so nerdctl should be probably case-sensitive too

File: pkg/imgutil/imgutil.go (hunk 4 above)
Comment (manugupt1) [reply pkg/imgutil/imgutil.go:373]: Every time there is a <none> image listed out; this Warning is printed out; I think we should not do it; now that we are adding more support for dangling images.

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**